test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -204,11 +204,11 @@ class LogicalDnsClusterTest : public Event::TestUsingSimulatedTime, public testi
   Network::DnsResolver::ResolveCb dns_callback_;
   NiceMock<ThreadLocal::MockInstance> tls_;
   Event::MockTimer* resolve_timer_;
-  std::shared_ptr<LogicalDnsCluster> cluster_;
   ReadyWatcher membership_updated_;
   ReadyWatcher initialized_;
   NiceMock<Runtime::MockLoader> runtime_;
   NiceMock<Event::MockDispatcher> dispatcher_;
+  std::shared_ptr<LogicalDnsCluster> cluster_;
```

Comment:
this change doesn't look necessary

Reply:
It is required. We now rely on the dispatcher to delete the ClusterInfo object. 
The dispatcher must be kept valid when scheduling ClusterInfo destruction. In the test case, the `cluster_` var keeps the last shared reference to `ClusterInfo`. That being said,  we need to destroy cluster_ before destroying the dispatcher.

Reply:
ok!